### PR TITLE
fix model leak issue

### PR DESCRIPTION
**Pull request by code-october**
#### 💻 变更类型 | Change Type

- [ ] feat    
- [x] fix    
- [ ] refactor    
- [ ] perf    
- [ ] style    
- [ ] test    
- [ ] docs    
- [ ] ci    
- [ ] chore    
- [ ] build    

#### 🔀 变更说明 | Description of Change

重新优化了模型可用性判断，避免模型泄露

#### 📝 补充信息 | Additional Information

受此 https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/issues/5881  启发，重新检查了模型禁用规则，发现存在重大问题，不仅 DISABLE_GPT4 无法生效，即使通过 CUSTOM_MODELS 限制了可用模型列表，依然无法真正限制模型请求（仅前端无法选中，但实际可以请求到 api 支持但 CUSTOM_MODELS 已限制的模型，如通过 curl 请求指定的模型）。

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function to check if a model is unavailable across multiple providers, enhancing error handling.

- **Bug Fixes**
	- Updated model availability checks across various APIs to return a 403 Forbidden response when a model is not available, improving clarity in error messaging.

- **Documentation**
	- Enhanced comments for the new function to clarify its parameters and return type.

- **Tests**
	- Added a comprehensive suite of unit tests for the new model availability function, ensuring correct behavior under various scenarios.

### CODE REVIEW

**Comment by Dogtiti** [comment]
加上单元测试~

**Comment by code-october** [comment]
已添加

File: test/model-available.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        expect(result).toBe(false);
+    });
+
+    test("test model will return false, which means the model is not available", () => {
+        const customModels = "-all,gpt-4o-mini";
+        const modelName = "gpt-4";
+        const providerNames = "OpenAI";
+        const result = isModelNotavailableInServer(customModels, modelName, providerNames);
+        expect(result).toBe(true);
+    });
```
[the remark this answers is not in the document]
Comment (code-october) [reply test/model-available.test.ts]: 已采纳

File: test/model-available.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    test("test custom model without setting provider", () => {
+        const customModels = "-all,mistral-large";
+        const modelName = "mistral-large";
+        const providerNames = modelName;
+        const result = isModelNotavailableInServer(customModels, modelName, providerNames);
+        expect(result).toBe(false);
+    });
+})
```
[the remark this answers is not in the document]
Comment (code-october) [reply test/model-available.test.ts]: 已采纳

**Review by Dogtiti — APPROVED** [review]

**Pull request merged**